tests/e2e/lib.sh

```diff
@@ -331,6 +335,40 @@ setup_podsecuritypolicies_config() {
     fi
 }
 
+# wait_for_collectors_to_be_operational() ensures that collector pods are able
+# to load kernel objects and create network connections.
+wait_for_collectors_to_be_operational() {
+    info "Will wait for collectors to reach a ready state"
+
+    local readiness_indicator="Successfully established GRPC stream for signals"
+    local start_time
+    start_time="$(date '+%s')"
+    local all_ready="false"
+    while [[ "$all_ready" == "false" ]]; do
+        all_ready="true"
+        for pod in $(kubectl -n stackrox get pods -l app=collector -o json | jq -r '.items[].metadata.name'); do
+            echo "Checking readiness of $pod"
+            if kubectl -n stackrox logs -c collector "$pod" | grep "$readiness_indicator" > /dev/null 2>&1; then
+                echo "ready"
+            else
+                info "$pod is not ready"
+                kubectl -n stackrox logs -c collector "$pod"
+                all_ready="false"
+                break
+            fi
+        done
+        if (( $(date '+%s') - start_time > 300 )); then
```

Comment:
[nit] You could create a variable to contain the timeout period and use that here and in the echo.